google-cloud-firestore/src/main/java/com/google/cloud/firestore/Firestore.java

```diff
@@ -45,10 +46,10 @@
   /**
    * Fetches the root collections that are associated with this Firestore database.
    *
-   * @return An ApiFuture that will be resolved with the list of collections.
+   * @return An Iterable that can be used to fetch all collections.
    */
   @Nonnull
-  ApiFuture<List<CollectionReference>> getCollections();
+  Iterable<CollectionReference> getCollections() throws ExecutionException, InterruptedException;
```

Comment:
I tried to throw a FirestoreException at first, but for that I need an IOException.

I updated the code to use the ApiExceptions code and added the ApiException to the list of declared exceptions. Am I correct to assume that this is the unchecked exception that we would throw most times?

Reply:
Hm, looks like it will mostly be a UncheckedExecutionException. That sounds like something we wouldn't want to add to the declared exceptions.

Is there a way we can throw a FirestoreException that doesn't require an IOException? The super class 'BaseHttpServiceException' seems to geared towards dealing with network problems.

Reply:
Ok, several things to say here:

* `FirestoreException` should extend `BaseGrpcServiceException`, not `BaseHttpServiceException`. 
* Relatedly, you should remove your dependency on google-cloud-core-http.
* The base exceptions accept `IOException` as a convenience, but the primary constructor is actually the one accepting `(String message, Throwable cause, int code, boolean retryable)`. Better for your case, you should actually use the constructor of `BaseGrpcServiceException` that accepts `ApiException`.
* You wouldn't need to declare that you're throwing `ApiException` since it is a `RuntimeException`; you can just get by with documenting it in a `@throws` tag in javadoc. Anyway, you should just wrap the `ApiException` with `FirestoreException` so that is moot.